src/components/site/tableOfContents/index.jsx

```diff
@@ -40,11 +46,23 @@ class TableOfContents extends React.PureComponent {
         label: link.title,
         to: `${this.TO_PREFIX}${link.slug}`,
         primaryText: link.title,
+        leftIcon: <Check />,
         active: this.setActivePage(link.id),
+        className: classnames(
+          'toc-link',
+          `link--page-id--${link.id}`,
+          this.setActivePage(link.id) ? 'link-active' : '',
+          this.checkQAProgress(link.id) ? 'qa-progress--complete' : ''
+        ),
       };
     });
   };
 
+  checkQAProgress = pageId => {
+    // add some fancy logic to reflect page answer state.
+    return +pageId === 5;
+  };
+
   getNavLinks = navLinks => [...this.routes, ...this.massageNavLinks(navLinks)];
 
   setActivePage = linkId => linkId === this.global.pageId;
```

Comment:
consider renaming this method to something like `isActivePage` or `checkActivePage` and reserving the `set` or `get` prefixes for methods that alter state or return something.  Your method is an evaluation, and it's just returning a boolean, a check, whereas, I think one would expect setActivePage to either change what page is active in state, or return "the active page."